src/main/java/uk/gov/ons/census/fwmt/jobservice/service/JobService.java

```diff
@@ -1,18 +1,41 @@
 package uk.gov.ons.census.fwmt.jobservice.service;
 
-import uk.gov.ons.census.fwmt.canonical.v1.CancelFieldWorkerJobRequest;
-import uk.gov.ons.census.fwmt.canonical.v1.CreateFieldWorkerJobRequest;
-import uk.gov.ons.census.fwmt.canonical.v1.UpdateFieldWorkerJobRequest;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
 import uk.gov.ons.census.fwmt.common.error.GatewayException;
+import uk.gov.ons.census.fwmt.events.component.GatewayEventManager;
+import uk.gov.ons.census.fwmt.jobservice.dto.rm.FieldworkFollowup;
+import uk.gov.ons.census.fwmt.jobservice.dto.tm.PutCaseRequest;
+import uk.gov.ons.census.fwmt.jobservice.service.comet.CometRestClient;
+import uk.gov.ons.census.fwmt.jobservice.service.mapper.SpgMapper;
 
-public interface JobService {
-  void createJob(CreateFieldWorkerJobRequest jobRequest) throws GatewayException;
+@Slf4j
+@Service
+public class JobService {
 
-  void cancelJob(CancelFieldWorkerJobRequest cancelRequest) throws GatewayException;
+  private final SpgMapper spgMapper;
+  private final CometRestClient cometRestClient;
+  private final GatewayEventManager gatewayEventManager;
 
-  void updateJob(UpdateFieldWorkerJobRequest updateRequest) throws GatewayException;
+  public JobService(SpgMapper spgMapper, CometRestClient cometRestClient, GatewayEventManager gatewayEventManager) {
+    this.spgMapper = spgMapper;
+    this.cometRestClient = cometRestClient;
+    this.gatewayEventManager = gatewayEventManager;
+  }
 
-  void convertAndSendCreate(CreateFieldWorkerJobRequest jobRequest) throws GatewayException;
+  public void handleMessage(FieldworkFollowup fieldworkFollowup) throws GatewayException {
+    PutCaseRequest putCase = map(fieldworkFollowup);
 
-  void convertAndSendCancel(CancelFieldWorkerJobRequest cancelJobRequest) throws GatewayException;
-}
\ No newline at end of file
+    cometRestClient.sendRequest(putCase, fieldworkFollowup.getCaseId());
+  }
+
+  public PutCaseRequest map(FieldworkFollowup fieldworkFollowup) {
+    if (fieldworkFollowup.getAddressType().equals("SPG")) {
+      return spgMapper.map(fieldworkFollowup);
+    } else {
+      // TODO proper error handling
```

Comment:
for now probably create a meaningful error event for use in log dashboards and throw a gateway exception.